src/Arblib.jl

```diff
@@ -41,6 +42,25 @@ macro libflint(function_name)
     return (:($function_name), Arb_jll.libflint)
 end
 
+const __isthreaded = Ref(false)
+
+function __init__()
+    __isthreaded[] = Base.get(ENV, "NEMO_THREADED", "") == "1"
+
+    if __isthreaded[]
+        ccall((:__gmp_set_memory_functions, :libgmp), Nothing,
```

Comment:
why do we need to set those memory functions? also is that really `0::Int` or rather `C_NULL::Ptr{Cvoid}`?

Reply:
I just copied the codes from `Nemo.jl` codebase, since I am not familiar with `gmp` or `flint`.

Reply:
@Joel-Dahne could you chip in?

Reply:
It works for me even if I don't set these memory functions, but I don't know what they do so it could depend on your system or on what you are doing for example. It actually works even without setting the `NEMO_THREADED` to `1` for me, so I don't even know what that part does...

It could maybe be that Nemo has these methods for historical reasons, they use Flint before `flint_jll` was a thing, and that they are not needed anymore.

How about we try to just not set these memory functions for now and hope that things work? And then we add it back if we notice any issues? I would like to avoid touching gmp things which I don't understand if we can avoid it.

Reply:
Are you suggesting we just wrap the `flint_set_num_threads` function and do nothing else?

Reply:
Exactly! At least it works on my Linux machine. I don't know what you run on but it would be good if you could test that it works as well. Also if we add the tests I propose below they will also be run on both Windows and Mac with the continuous integration.